hubblestack/extmods/hubble_mods/ssl_certificate.py

```diff
@@ -0,0 +1,315 @@
+            cert_details = cert_file.read()
+    except IOError as e:
+        log.error('File not found: {0}. Error: {1}'.format(cert_file_path, e))
+        return None
+    return cert_details
+
+
+def _parse_cert(cert, host, port, path):
+    """
+    load the certificate using OpenSSL and parse needed params.
+    """
+    log.debug("Parsing the fetched certificate")
+    cert_details = {}
+    try:
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        cert_details['ssl_src_port'] = str(port)
+        cert_details['ssl_src_host'] = str(host)
+        cert_details['ssl_src_path'] = str(path)
+        if x509.get_issuer():
+            issuer_components = _format_components(x509.get_issuer())
+            cert_details['ssl_issuer_common_name'] = issuer_components.get('CN', "None")
+        if x509.get_subject():
+            subject_components = _format_components(x509.get_subject())
+            cert_details['ssl_subject_country'] = subject_components.get('C', "None")
+            cert_details['ssl_subject_organisation'] = subject_components.get('O', "None")
+            cert_details['ssl_subject_organisation_unit'] = subject_components.get('OU', "None")
+            cert_details['ssl_subject_common_name'] = subject_components.get('CN', "None")
+        not_after = datetime.strptime(x509.get_notAfter().decode('utf-8'), "%Y%m%d%H%M%SZ")
+        not_before = datetime.strptime(x509.get_notBefore().decode('utf-8'), "%Y%m%d%H%M%SZ")
+        has_expired = x509.has_expired()
+        cert_details['ssl_cert_version'] = str(x509.get_version())
+        cert_details['ssl_has_expired'] = True if has_expired == 1 else False
+        cert_details['ssl_serial_number'] = str(x509.get_serial_number())
+        cert_details['ssl_end_time'] = str(not_after)
+        cert_details['ssl_start_time'] = str(not_before)
+        cert_details['ssl_signature_algorithm'] = str(x509.get_signature_algorithm())
+        cert_details['ssl_cert_pem'] = str(cert)
+        cert_details['ssl_subject_alternative_names'] = _get_certificate_san(x509)
+    except Exception as e:
+        cert_details['error'] = "An error occurred while parsing certificate - {0}".format(e)
+    return cert_details
+
+
+def _format_components(x509name):
+    items = {}
+    for item in x509name.get_components():
+        items[item[0]] = item[1]
+    return items
+
+
+def _get_certificate_san(x509cert):
+    san = ''
+    trimmed_san_list = []
+    try:
+        ext_count = x509cert.get_extension_count()
+        for i in range(0, ext_count):
+            ext = x509cert.get_extension(i)
+            if 'subjectAltName' in str(ext.get_short_name()):
+                san = ext.__str__()
+        san_list = san.split(',')
+        for san in san_list:
+            trimmed_san = san.lstrip()
+            trimmed_san_list.append(trimmed_san)
+    except Exception as e:
+        message = "FDG ssl_certificate couldn't fetch SANs: {0}".format(e)
```

Comment:
Remove fdg.
And, exception is silently getting passed. Please check.

Reply:
removed fdg.
Here exception is expected to be passed.

Reply:
same behavior expected. not modifying